extensions/opc/src/fetchopc.cpp

```diff
@@ -0,0 +1,235 @@
+    logger_->log_trace("FetchOPCProcessor::onSchedule");
+
+    translatedNodeIDs_.clear();  // Path might has changed during restart
+
+    BaseOPCProcessor::onSchedule(context, factory);
+
+    if(!configOK_) {
+      return;
+    }
+
+    configOK_ = false;
+
+    std::string value;
+    context->getProperty(NodeID.getName(), nodeID_);
+    context->getProperty(NodeIDType.getName(), value);
+
+    maxDepth_ = 0;
+    context->getProperty(MaxDepth.getName(), maxDepth_);
+
+    if (value == "String") {
+      idType_ = opc::OPCNodeIDType::String;
+    } else if (value == "Int") {
+      idType_ = opc::OPCNodeIDType::Int;
+    } else if (value == "Path") {
+      idType_ = opc::OPCNodeIDType::Path;
+    } else {
+      // Where have our validators gone?
+      logger_->log_error("%s is not a valid node ID type!", value.c_str());
+    }
+
+    if(idType_ == opc::OPCNodeIDType::Int) {
+      try {
+        int t = std::stoi(nodeID_);
+      } catch(...) {
+        logger_->log_error("%s cannot be used as an int type node ID", nodeID_.c_str());
+        return;
+      }
+    }
+    if(idType_ != opc::OPCNodeIDType::Path) {
+      if(!context->getProperty(NameSpaceIndex.getName(), nameSpaceIdx_)) {
+        logger_->log_error("%s is mandatory in case %s is not Path", NameSpaceIndex.getName().c_str(), NodeIDType.getName().c_str());
+        return;
+      }
+    }
+
+    configOK_ = true;
+  }
+
+  void FetchOPCProcessor::onTrigger(const std::shared_ptr<core::ProcessContext> &context, const std::shared_ptr<core::ProcessSession> &session){
+    if(!configOK_) {
+      logger_->log_error("This processor was not configured properly, yielding. Please check for previous errors in the logs!");
+      yield();
+      return;
+    }
+
+    logger_->log_trace("FetchOPCProcessor::onTrigger");
+
+    std::unique_lock<std::mutex> lock(onTriggerMutex_, std::try_to_lock);
+    if(!lock.owns_lock()){
+      logger_->log_warn("processor was triggered before previous listing finished, configuration should be revised!");
+      return;
+    }
+
+    if (!reconnect()) {
+      yield();
+      return;;
+    }
+
+    nodesFound_ = 0;
+    variablesFound_ = 0;
+
+    std::function<opc::nodeFoundCallBackFunc> f = std::bind(&FetchOPCProcessor::nodeFoundCallBack, this, std::placeholders::_1, std::placeholders::_2, std::placeholders::_3, context, session);
+    if(idType_ != opc::OPCNodeIDType::Path) {
+      UA_NodeId myID;
+      myID.namespaceIndex = nameSpaceIdx_;
+      if(idType_ == opc::OPCNodeIDType::Int) {
+        myID.identifierType = UA_NODEIDTYPE_NUMERIC;
+        myID.identifier.numeric = std::stoi(nodeID_);
+      } else if (idType_ == opc::OPCNodeIDType::String) {
+        myID.identifierType = UA_NODEIDTYPE_STRING;
+        myID.identifier.string = UA_STRING_ALLOC(nodeID_.c_str());
+      }
+      connection_->traverse(myID, f, "", maxDepth_);
```

Comment:
traverse's maxDepth parameter is an uint32_t but we are supplying an uint64_t here.

Reply:
Good spot! Changed the signature of traverse.
It happened because of the lack of uint32_t getValue specialisation, which made me change the type of maxDepth.